src/varcache.c

```diff
@@ -99,26 +167,48 @@ static int apply_var(PktBuf *pkt, const char *key,
 	if (strcasecmp(cval->str, sval->str) == 0)
 		return 0;
 
-	/* the string may have been taken from startup pkt */
-	if (!pg_quote_literal(qbuf, cval->str, sizeof(qbuf)))
+	/* parameters that are marked GUC_LIST_QUOTE are returned already fully quoted
+	 * re-quoting them using pg_quote_literal will result in malformed values. */
+	if (variable_is_guc_list_quote(key)) {
+		/* zero length elements of the form "" should be specially handled.*/
+		if (strlen(cval->str) == 2 && strcmp(cval->str,"\"\"") == 0) {
```

Comment:
No need to check strlen beforehand
```suggestion
		if (strcmp(cval->str,"\"\"") == 0) {
```